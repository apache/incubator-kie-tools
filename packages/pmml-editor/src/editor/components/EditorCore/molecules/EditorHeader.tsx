/*
 * Copyright 2020 Red Hat, Inc. and/or its affiliates.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *       http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
import { ModelTitle } from "../atoms";
import * as React from "react";
import { Split, SplitItem } from "@patternfly/react-core/dist/js/layouts/Split";
import DataDictionaryHandler from "../../DataDictionary/DataDictionaryHandler/DataDictionaryHandler";
import { OutputsHandler } from "../../Outputs/organisms";
import { FieldName, MiningSchema, Output, OutputField } from "@kie-tools/pmml-editor-marshaller";
import MiningSchemaHandler from "../../MiningSchema/MiningSchemaHandler/MiningSchemaHandler";
import "./EditorHeader.scss";

interface EditorHeaderViewerProps {
  modelName: string;
  modelIndex: number;
}

interface EditorHeaderEditorProps extends EditorHeaderViewerProps {
  output?: Output;
  miningSchema?: MiningSchema;
  validateOutputFieldName: (index: number | undefined, name: FieldName) => boolean;
  deleteOutputField: (index: number) => void;
  commitOutputField: (index: number | undefined, outputField: OutputField) => void;
  commitModelName: (modelName: string) => void;
}

type EditorHeaderProps = EditorHeaderViewerProps | EditorHeaderEditorProps;

const isEditor = (props: EditorHeaderProps): props is EditorHeaderEditorProps => {
  return (props as EditorHeaderEditorProps).modelIndex !== undefined;
};

export const EditorHeader = (props: EditorHeaderProps) => {
  const { modelName, modelIndex } = props;

  if (isEditor(props)) {
    const { miningSchema, output, validateOutputFieldName, deleteOutputField, commitOutputField, commitModelName } =
      props;

    return (
      <Split hasGutter={true} className={"editorHeader"}>
        <SplitItem className={"editorHeader__modelName"}>
          <ModelTitle modelName={modelName} commitModelName={commitModelName} />
        </SplitItem>
        <SplitItem isFilled={true} data-ouia-component-type="filler" />
        <SplitItem>
          <DataDictionaryHandler />
        </SplitItem>
        <SplitItem>
          <MiningSchemaHandler miningSchema={miningSchema} modelIndex={modelIndex} />
        </SplitItem>
        <SplitItem>
          <OutputsHandler
            modelIndex={modelIndex}
            output={output}
            miningSchema={miningSchema}
            validateOutputFieldName={validateOutputFieldName}
            deleteOutputField={deleteOutputField}
            commitOutputField={commitOutputField}
          />
        </SplitItem>
      </Split>
    );
  } else {
    return (
      <Split hasGutter={true} className={"editorHeader"}>
        <SplitItem isFilled={true} className={"editorHeader--modelName"}>
          <ModelTitle modelName={modelName} />
        </SplitItem>
      </Split>
    );
  }
};
