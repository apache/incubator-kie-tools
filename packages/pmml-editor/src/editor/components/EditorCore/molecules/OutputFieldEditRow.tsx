/*
 * Copyright 2020 Red Hat, Inc. and/or its affiliates.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *       http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
import * as React from "react";
import { useMemo, useState } from "react";
import { Flex, FlexItem, FormGroup, Select, SelectOption, SelectVariant, TextInput } from "@patternfly/react-core";
import "../organisms/OutputFieldsTable.scss";
import { FieldName, OutputField } from "@kogito-tooling/pmml-editor-marshaller";
import { OutputFieldRowEditModeAction, OutputLabelsEditMode } from "../atoms";
import { ExclamationCircleIcon } from "@patternfly/react-icons";

interface OutputFieldEditRowProps {
  activeOutputFieldIndex: number | undefined;
  activeOutputField: OutputField;
  setActiveOutputField: (_output: OutputField) => void;
  validateOutputName: (name: string | undefined) => boolean;
  viewExtendedProperties: () => void;
  onCommit: () => void;
  onCancel: () => void;
}

const dataTypes = [
  "string",
  "integer",
  "float",
  "double",
  "boolean",
  "date",
  "time",
  "dateTime",
  "dateDaysSince[0]",
  "dateDaysSince[1960]",
  "dateDaysSince[1970]",
  "dateDaysSince[1980]",
  "timeSeconds",
  "dateTimeSecondsSince[0]",
  "dateTimeSecondsSince[1960]",
  "dateTimeSecondsSince[1970]",
  "dateTimeSecondsSince[1980]"
];

export const OutputFieldEditRow = (props: OutputFieldEditRowProps) => {
  const {
    activeOutputFieldIndex,
    activeOutputField,
    setActiveOutputField,
    validateOutputName,
    viewExtendedProperties,
    onCommit,
    onCancel
  } = props;

  const [isTypeSelectOpen, setIsTypeSelectOpen] = useState(false);
  const typeToggle = (isOpen: boolean) => {
    setIsTypeSelectOpen(isOpen);
  };

  const onSelectType = (event: any, selection: any, isPlaceholder: boolean) => {
    setActiveOutputField({ ...activeOutputField, dataType: isPlaceholder ? undefined : selection });
    setIsTypeSelectOpen(false);
  };

  const isValidName = useMemo(() => validateOutputName(activeOutputField.name.toString()), [
    activeOutputFieldIndex,
    activeOutputField
  ]);

  return (
    <article className={`output-item output-item-n${activeOutputFieldIndex}`}>
      <Flex alignItems={{ default: "alignItemsCenter" }} style={{ height: "100%" }}>
        <FlexItem>
          <FormGroup
            label="Name"
            fieldId="output-name-helper"
            helperTextInvalid="Name must be unique and present."
            helperTextInvalidIcon={<ExclamationCircleIcon />}
            validated={isValidName ? "default" : "error"}
            style={{ width: "16em" }}
          >
            <TextInput
              type="text"
              id="output-name"
              name="output-name"
              aria-describedby="output-name-helper"
              value={activeOutputField.name.toString()}
              validated={isValidName ? "default" : "error"}
              autoFocus={true}
              onChange={e => {
                setActiveOutputField({ ...activeOutputField, name: e as FieldName });
              }}
            />
          </FormGroup>
        </FlexItem>
        <FlexItem>
          <FormGroup label="Data type" fieldId="output-dataType-helper" style={{ width: "12em" }}>
            <Select
              id="output-dataType"
              name="output-dataType"
              aria-label="Select Input"
              aria-describedby="output-dataType-helper"
              variant={SelectVariant.single}
              onToggle={typeToggle}
              onSelect={onSelectType}
              selections={activeOutputField.dataType}
              isOpen={isTypeSelectOpen}
              placeholder="Type"
              menuAppendTo={"parent"}
            >
              {dataTypes.map((dt, _index) => (
                <SelectOption key={_index} value={dt} />
              ))}
            </Select>
          </FormGroup>
        </FlexItem>
        <FlexItem>
          <FormGroup label="Properties" fieldId="output-labels-helper">
            <OutputLabelsEditMode
              activeOutputField={activeOutputField}
              setActiveOutputField={setActiveOutputField}
              viewExtendedProperties={viewExtendedProperties}
            />
          </FormGroup>
        </FlexItem>
        <FlexItem align={{ default: "alignRight" }}>
          <OutputFieldRowEditModeAction onCommit={onCommit} onCancel={onCancel} disableCommit={!isValidName} />
        </FlexItem>
      </Flex>
    </article>
  );
};
