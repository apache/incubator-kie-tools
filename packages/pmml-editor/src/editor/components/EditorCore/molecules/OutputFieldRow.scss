.output-item {
  min-height: 70px;
  border: 1px solid var(--pf-global--BorderColor--100);
  padding: 1em;
  background-color: #ffffff;
  margin-bottom: 1em;
  user-select: none;
  // write the following better
  span,
  strong {
    user-select: text;
  }
  &:last-child {
    margin-bottom: 0;
  }
  &__type-label {
    text-transform: capitalize;
  }
  &__type-select {
    &__option,
    .pf-c-select__toggle {
      text-transform: capitalize;
    }
  }
}
