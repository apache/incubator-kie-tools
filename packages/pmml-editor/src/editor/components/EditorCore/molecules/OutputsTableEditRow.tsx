/*
 * Copyright 2020 Red Hat, Inc. and/or its affiliates.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *       http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
import * as React from "react";
import { useMemo, useState } from "react";
import { Flex, FlexItem, FormGroup, Select, SelectOption, SelectVariant, TextInput } from "@patternfly/react-core";
import "../organisms/OutputsTable.scss";
import { FieldName, OutputField } from "@kogito-tooling/pmml-editor-marshaller";
import { OutputLabelsEditMode, OutputsTableEditModeAction } from "../atoms";
import { ExclamationCircleIcon } from "@patternfly/react-icons";

interface OutputsTableEditRowProps {
  activeOutputFieldIndex: number | undefined;
  activeOutputField: OutputField;
  setActiveOutputField: (_output: OutputField) => void;
  validateOutputName: (name: string | undefined) => boolean;
  viewExtendedProperties: () => void;
  onCommit: () => void;
  onCancel: () => void;
}

const dataTypes = [
  "string",
  "integer",
  "float",
  "double",
  "boolean",
  "date",
  "time",
  "dateTime",
  "dateDaysSince[0]",
  "dateDaysSince[1960]",
  "dateDaysSince[1970]",
  "dateDaysSince[1980]",
  "timeSeconds",
  "dateTimeSecondsSince[0]",
  "dateTimeSecondsSince[1960]",
  "dateTimeSecondsSince[1970]",
  "dateTimeSecondsSince[1980]"
];

export const OutputsTableEditRow = (props: OutputsTableEditRowProps) => {
  const {
    activeOutputFieldIndex,
    activeOutputField,
    setActiveOutputField,
    validateOutputName,
    viewExtendedProperties,
    onCommit,
    onCancel
  } = props;

  const [isTypeSelectOpen, setIsTypeSelectOpen] = useState(false);
  const typeToggle = (isOpen: boolean) => {
    setIsTypeSelectOpen(isOpen);
  };

  const onSelectType = (event: any, selection: any, isPlaceholder: boolean) => {
    setActiveOutputField({ ...activeOutputField, dataType: isPlaceholder ? undefined : selection });
    setIsTypeSelectOpen(false);
  };

  const isValidName = useMemo(() => validateOutputName(activeOutputField.name.toString()), [
    activeOutputFieldIndex,
    activeOutputField
  ]);

  return (
    <article className={`output-item output-item-n${activeOutputFieldIndex}`}>
      <Flex alignItems={{ default: "alignItemsCenter" }} style={{ height: "100%" }}>
        <FlexItem>
          <FormGroup
            fieldId="output-name-helper"
            helperText="Please provide a name for the Output Field."
            helperTextInvalid="Name must be unique and present."
            helperTextInvalidIcon={<ExclamationCircleIcon />}
            validated={isValidName ? "default" : "error"}
            style={{ width: "12em" }}
          >
            <TextInput
              type="text"
              id="output-name"
              name="output-name"
              aria-describedby="output-name-helper"
              value={activeOutputField.name.toString()}
              validated={isValidName ? "default" : "error"}
              autoFocus={true}
              onChange={e => {
                setActiveOutputField({ ...activeOutputField, name: e as FieldName });
              }}
            />
          </FormGroup>
        </FlexItem>
        <FlexItem>
          <FormGroup
            fieldId="output-dataType-helper"
            helperText="Specifies the data type for the output field."
            style={{ width: "12em" }}
          >
            <Select
              id="output-dataType"
              name="output-dataType"
              aria-label="Select Input"
              aria-describedby="output-dataType-helper"
              variant={SelectVariant.single}
              onToggle={typeToggle}
              onSelect={onSelectType}
              selections={activeOutputField.dataType}
              isOpen={isTypeSelectOpen}
              placeholder="Type"
              menuAppendTo={"parent"}
            >
              {dataTypes.map((dt, _index) => (
                <SelectOption key={_index} value={dt} />
              ))}
            </Select>
          </FormGroup>
        </FlexItem>
        <FlexItem>
          <OutputLabelsEditMode
            activeOutputField={activeOutputField}
            setActiveOutputField={setActiveOutputField}
            viewExtendedProperties={viewExtendedProperties}
          />
        </FlexItem>
        <FlexItem align={{ default: "alignRight" }}>
          <OutputsTableEditModeAction onCommit={onCommit} onCancel={onCancel} disableCommit={false} />
        </FlexItem>
      </Flex>
    </article>
  );
};
