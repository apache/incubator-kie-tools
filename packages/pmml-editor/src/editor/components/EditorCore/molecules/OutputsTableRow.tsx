/*
 * Copyright 2020 Red Hat, Inc. and/or its affiliates.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *       http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
import * as React from "react";
import { Flex, FlexItem, Label } from "@patternfly/react-core";
import { OutputField } from "@kogito-tooling/pmml-editor-marshaller";
import { OutputLabels, OutputsTableAction } from "../atoms";
import "./OutputsTableRow.scss";

interface OutputsTableRowProps {
  index: number;
  output: OutputField;
  onEdit: () => void;
  onDelete: () => void;
  isDisabled: boolean;
}

export const OutputsTableRow = (props: OutputsTableRowProps) => {
  const { index, output, onEdit, onDelete, isDisabled } = props;

  return (
    <article className={`output-item output-item-n${index}`}>
      <Flex alignItems={{ default: "alignItemsCenter" }} style={{ height: "100%" }}>
        <FlexItem>
          <strong>{output.name}</strong>
        </FlexItem>
        <FlexItem>
          <Label color="blue" className="output-item__type-label">
            {output.dataType}
          </Label>
        </FlexItem>
        <FlexItem>
          <OutputLabels output={output} />
        </FlexItem>
        <FlexItem align={{ default: "alignRight" }}>
          <OutputsTableAction onEdit={() => onEdit()} onDelete={() => onDelete()} disabled={isDisabled} />
        </FlexItem>
      </Flex>
    </article>
  );
};
