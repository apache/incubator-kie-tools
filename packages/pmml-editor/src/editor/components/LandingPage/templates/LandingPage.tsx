/*
 * Copyright 2020 Red Hat, Inc. and/or its affiliates.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *       http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
import * as React from "react";
import { useCallback, useMemo, useState } from "react";
import { Gallery, GalleryItem, PageSection, PageSectionVariants } from "@patternfly/react-core";
import { EmptyStateNoModels } from "../organisms";
import { v4 as uuid } from "uuid";
import { Model, PMML } from "@kogito-tooling/pmml-editor-marshaller";
import { useDispatch, useSelector } from "react-redux";
import { getModelName, isSupportedModelType } from "../../..";
import { ActionSelector, ModelCard } from "../molecules";
import { Actions } from "../../../reducers";
import { Header } from "../../Header/molecules";
import { useHistory } from "react-router";

interface LandingPageProps {
  path: string;
}

export const LandingPage = (props: LandingPageProps) => {
  const history = useHistory();
  const dispatch = useDispatch();

  const [filter, setFilter] = useState("");
  const [showUnsupportedModels, setShowUnsupportedModels] = useState(true);
  const hasUnsupportedModels = useMemo(() => (models ?? []).find(model => !isSupportedModelType(model)) !== undefined, [
    models
  ]);

  const models: Model[] | undefined = useSelector<PMML, Model[] | undefined>((state: PMML) => state.models);
  const hasUnsupportedModels = useMemo(() => (models ?? []).find(model => !isSupportedModelType(model)) !== undefined, [
    models
  ]);

  const filterModels = useCallback((): Model[] => {
    const _lowerCaseFilter = filter.toLowerCase();
    const _filteredModels = models?.filter((_model: Model) => {
      const _modelName = getModelName(_model);
      const _isSupportedModelType = isSupportedModelType(_model);
      const _nameMatch = _modelName === undefined || _modelName.toLowerCase().includes(_lowerCaseFilter);
      const _supportMatch = showUnsupportedModels || _isSupportedModelType;
      return _nameMatch && _supportMatch;
    });
    return _filteredModels ?? [];
  }, [filter, showUnsupportedModels, models]);

  const filteredModels: Model[] = useMemo(() => filterModels(), [filter, showUnsupportedModels, models]);

  const goToModel = useCallback(
    (index: number) => {
      history.push({
        pathname: "editor/" + index
      });
    },
    [history]
  );

  return (
    <div data-testid="landing-page">
      <PageSection variant={PageSectionVariants.light}>
        <Header title={props.path} />
        <LandingPageToolbar
          setFilter={setFilter}
          hasUnsupportedModels={hasUnsupportedModels}
          showUnsupportedModels={showUnsupportedModels}
          setShowUnsupportedModels={setShowUnsupportedModels}
        />
      </PageSection>

      <PageSection isFilled={true}>
        <section>
          {filteredModels.length > 0 && (
            <Gallery hasGutter={true}>
              {filteredModels.map(model => {
                //model should always be a member of models at this point.
                const index: number | undefined = models?.indexOf(model);
                const modelName: string = getModelName(model);
                const modelType: ModelType = getModelType(model);

                return (
                  <GalleryItem key={uuid()} data-testid="landing-page__model-card">
                    <ModelCard
                      index={index}
                      modelName={modelName}
                      modelType={modelType}
                      onClick={goToModel}
                      onDelete={(_model: Model) => {
                        if (window.confirm(`Delete Model "${modelName}"?`)) {
                          dispatch({
                            type: Actions.DeleteModel,
                            payload: {
                              modelIndex: index
                            }
                          });
                        }
                      }}
                    />
                  </GalleryItem>
                );
              })}
            </Gallery>
          )}
          {filteredModels.length === 0 && (
            <EmptyStateNoModels
              createModel={() => {
                window.alert("TODO: Create Model");
              }}
            />
          )}
        </section>
      </PageSection>
    </div>
  );
};
