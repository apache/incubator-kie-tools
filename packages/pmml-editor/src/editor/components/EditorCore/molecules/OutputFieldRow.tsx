/*
 * Copyright 2020 Red Hat, Inc. and/or its affiliates.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *       http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
import * as React from "react";
import { Label, Split, SplitItem } from "@patternfly/react-core";
import { OutputField } from "@kogito-tooling/pmml-editor-marshaller";
import { OutputFieldRowAction, OutputLabels } from "../atoms";
import "./OutputFieldRow.scss";

interface OutputFieldRowProps {
  activeOutputFieldIndex: number;
  activeOutputField: OutputField;
  onEditOutputField: () => void;
  onDeleteOutputField: () => void;
  isDisabled: boolean;
}

export const OutputFieldRow = (props: OutputFieldRowProps) => {
  const { activeOutputFieldIndex, activeOutputField, onEditOutputField, onDeleteOutputField, isDisabled } = props;

  return (
    <article className={`output-item output-item-n${activeOutputFieldIndex}`}>
      <Split hasGutter={true} style={{ height: "100%" }}>
        <SplitItem>
          <strong>{activeOutputField.name}</strong>
        </SplitItem>
        <SplitItem isFilled={true}>
          <Label color="blue" className="output-item__type-label">
            {activeOutputField.dataType}
          </Label>
          <OutputLabels activeOutputField={activeOutputField} />
        </SplitItem>
        <SplitItem>
          <OutputFieldRowAction
            onEditDataField={onEditOutputField}
            onDeleteDataField={onDeleteOutputField}
            disabled={isDisabled}
          />
        </SplitItem>
      </Split>
    </article>
  );
};
