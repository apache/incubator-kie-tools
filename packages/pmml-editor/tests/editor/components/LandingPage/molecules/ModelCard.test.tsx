/*
 * Copyright 2020 Red Hat, Inc. and/or its affiliates.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *       http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
import { render } from "@testing-library/react";
import * as React from "react";
import { Scorecard } from "@kogito-tooling/pmml-editor-marshaller";
import { ModelCard } from "@kogito-tooling/pmml-editor/dist/editor/components/LandingPage/molecules";

describe("ModelCard", () => {
  test("render::Basics", () => {
    const onClick = jest.fn((index: number) => null);
    const onDelete = jest.fn((index: number) => null);

    const { getByTestId } = render(
      <ModelCard index={0} modelName={"Name"} modelType={"Scorecard"} onClick={onClick} onDelete={onDelete} />
    );
    expect(getByTestId("model-card")).toMatchSnapshot();

    const element: HTMLElement = getByTestId("model-card__model-type");
    expect(element).toBeInTheDocument();
    expect(element).toHaveTextContent("Scorecard");
  });

  test("Delete::click", () => {
    const onClick = jest.fn((index: number) => null);
    const onDelete = jest.fn((index: number) => null);

    const { getByTestId } = render(
      <ModelCard index={0} modelName={"Name"} modelType={"Scorecard"} onClick={onClick} onDelete={onDelete} />
    );
    const element: HTMLElement = getByTestId("model-card__delete");
    expect(element).toBeInTheDocument();
    expect(element).toBeInstanceOf(HTMLButtonElement);

    (element as HTMLButtonElement).click();
    expect(onDelete).toBeCalledTimes(1);
    expect(onDelete).toBeCalledWith(0);
    expect(onClick).not.toBeCalled();
  });

  test("Click::click", () => {
    const onClick = jest.fn((index: number) => null);
    const onDelete = jest.fn((index: number) => null);

    const { getByTestId } = render(
      <ModelCard index={0} modelName={"Name"} modelType={"Scorecard"} onClick={onClick} onDelete={onDelete} />
    );
    const element: HTMLElement = getByTestId("model-card");
    expect(element).toBeInTheDocument();
    expect(element).toBeInstanceOf(HTMLElement);

    (element as HTMLElement).click();
    expect(onClick).toBeCalledTimes(1);
    expect(onClick).toBeCalledWith(0);
    expect(onDelete).not.toBeCalled();
  });
});
