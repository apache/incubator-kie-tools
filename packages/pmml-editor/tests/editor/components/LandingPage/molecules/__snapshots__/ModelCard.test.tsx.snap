// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ModelCard render::Basics 1`] = `
<article
  class="pf-c-card pf-m-hoverable model-card"
  data-ouia-component-id="OUIA-Generated-Card-1"
  data-ouia-component-type="PF4/Card"
  data-ouia-safe="true"
  data-testid="model-card"
  id=""
>
  <div
    class="pf-c-card__header"
  >
    <div
      class=""
    >
      <img
        alt="Scorecard"
        class="model-card__icon"
        data-testid="model-card__icon"
        src="images/card-icon-scorecard.svg"
      />
    </div>
  </div>
  <div
    class="pf-c-card__title model-card__title"
  >
    <span
      data-testid="model-card__title"
    >
      Name
    </span>
  </div>
   
  <div
    class="pf-c-card__body"
  >
    <div
      data-testid="model-card__model-type"
    >
      Scorecard
    </div>
  </div>
  <div
    class="pf-c-card__footer"
  >
    <button
      aria-disabled="false"
      class="pf-c-button pf-m-primary"
      data-ouia-component-id="OUIA-Generated-Button-primary-1"
      data-ouia-component-type="PF4/Button"
      data-ouia-safe="true"
      data-testid="model-card__delete"
      type="button"
    >
      Delete
    </button>
  </div>
</article>
`;
